altapay.php

```diff
@@ -0,0 +1,2353 @@
+		`refunded` int(10) NOT NULL DEFAULT 0,
+		PRIMARY KEY (`altapay_payment_id`,`product_id`)
+		) ENGINE=" . _MYSQL_ENGINE_ . " DEFAULT CHARSET=utf8");
+        }
+
+        if (Db::getInstance()->Execute("SELECT 1 FROM `" . _DB_PREFIX_ . "valitor_saved_credit_card`")) {
+
+            $sql = "RENAME TABLE  `" . _DB_PREFIX_ . "valitor_saved_credit_card`  TO `" . _DB_PREFIX_ . "altapay_saved_credit_card`  ";
+            Db::getInstance()->Execute($sql);
+
+        } else {
+            Db::getInstance()->Execute("CREATE TABLE IF NOT EXISTS `" . _DB_PREFIX_ . "altapay_saved_credit_card` (
+		`id` mediumint(9) NOT NULL AUTO_INCREMENT,
+		`time` datetime DEFAULT '0000-00-00 00:00:00' NOT NULL,
+		`userID` varchar(200) DEFAULT '' NOT NULL,
+		`cardBrand` varchar(200) DEFAULT '' NOT NULL,
+		`creditCardNumber` varchar(200) DEFAULT '' NOT NULL,
+		`cardExpiryDate` varchar(200) DEFAULT '' NOT NULL,
+		`ccToken` varchar(200) DEFAULT '' NOT NULL,
+		PRIMARY KEY  (`id`)
+		) ENGINE=" . _MYSQL_ENGINE_ . "  DEFAULT CHARSET=utf8 AUTO_INCREMENT=1");
+        }
+
+
+        if (!Db::getInstance()->Execute("ALTER TABLE `" . _DB_PREFIX_ . "altapay_orderlines`
+            MODIFY `product_id` varchar(36) NOT NULL")) {
+            $this->context->controller->errors[] = Db::getInstance()->getMsgError();
+            return false;
+        }
+
+        /* This table captures the payment information */
+        if (Db::getInstance()->Execute("SELECT 1 FROM `" . _DB_PREFIX_ . "valitor_cartInfo`")) {
+
+            $sql = "RENAME TABLE  `" . _DB_PREFIX_ . "valitor_cartInfo`  TO `" . _DB_PREFIX_ . "altapay_cartInfo`  ";
+            Db::getInstance()->Execute($sql);
+
+        } else {
+            Db::getInstance()->Execute('CREATE TABLE IF NOT EXISTS `' . _DB_PREFIX_ . 'altapay_cartInfo` (
+			`id_cart` int(10) unsigned NOT NULL,
+			`productDetails` varchar(255) NOT NULL,
+			`date_add` varchar(50) NOT NULL,
+			PRIMARY KEY (`id_cart`)
+		) ENGINE=' . _MYSQL_ENGINE_ . '  DEFAULT CHARSET=utf8');
+        }
+
+
+        $this->createOrderState();
+
+        return true;
+    }
+
+    /**
+     * Create a new order state
+     */
+    public function createOrderState()
+    {
+        if (!Configuration::get('ALTAPAY_OS_PENDING')) {
+            $orderState = new OrderState();
+            $orderState->name = array();
+
+            foreach (Language::getLanguages() as $language) {
+                $orderState->name[$language['id_lang']] = 'Awaiting payment processing';
+            }
+
+            $orderState->color = '#ffff5a';
+
+            $orderState->logable = false;
+            $orderState->invoice = false;
+            $orderState->hidden = false;
+            $orderState->send_email = false;
+            $orderState->shipped = false;
+            $orderState->paid = false;
+            $orderState->delivery = false;
+
+            if ($orderState->add()) {
+                $source = dirname(__FILE__) . '/views/img/os_pending.gif';
```

Comment:
use __DIR__ for current file dir:
```suggestion
                $source = __DIR__ . '/views/img/os_pending.gif';
```